node-launchpad/src/system.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use color_eyre::eyre::ContextCompat;
+use color_eyre::Result;
+use std::env;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Command;
+use sysinfo::Disks;
+
+// Tries to get the default (drive name, mount point) of the current executable
+// to be used as the default drive
+pub fn get_default_mount_point() -> (String, String) {
+    // Create a new System instance
+    let disks = Disks::new_with_refreshed_list();
+
+    // Get the current executable path
+    let exe_path = env::current_exe().expect("Failed to get current executable path");
+
+    // Iterate over the disks and find the one that matches the executable path
+    for disk in disks.list() {
+        if exe_path.starts_with(disk.mount_point()) {
+            return (
+                disk.name().to_string_lossy().into(),
+                disk.mount_point().to_string_lossy().into_owned(),
+            );
+        }
+    }
+    // If no matching disk is found, return an empty string or handle the error as needed
+    (String::new(), String::new())
```

Comment:
If empty string is returned here, how are things handled at the other components? Should we be more explicit and return a `None` incase if something doesn't work out instead of an empty string. It would be cleaner to handle.